app/Models/Multiplayer/Room.php

```diff
@@ -478,6 +478,16 @@ private function assertValidStartGame()
             throw new InvariantException("'ends_at' must be at least 30 minutes after 'starts_at'");
         }
 
+        // TODO: configurable duration (same for min and max durations)?
+        if (!$this->user->isSupporter() && $this->starts_at->addDays(14)->lt($this->ends_at)) {
+            throw new InvariantException(osu_trans('multiplayer.room.errors.duration_require_supporter'));
+        }
+
+        // see https://github.com/ppy/osu/pull/16024/files#diff-d5f8d0eb0eac5cfd6d2f486d34c4168e036b83b622c7a3c5bfce5205d67bf52bR327-R330
+        if ($this->starts_at->adddays(63)->lt($this->ends_at)) {
+            throw new InvariantException(osu_trans('multiplayer.room.errors.duration_too_long'));
+        }
```

Comment:
- almost none of other supporter/non-supporter checks do separate check like this at the moment (they all simply call relevant `maxThing()` function in User model)
- the message for non-supporter is kind of a lie if it's more than 3 months
- the second `addDays` has lowercase `d` for `Days`
- probably micro optimization but all of those date attribute accesses re-create the object every time

Reply:
`addDays` mutates the instance

Reply:
ends_at doesn't seem to be mutated anywhere though